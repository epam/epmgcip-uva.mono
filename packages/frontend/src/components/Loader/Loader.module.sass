@use 'src/colors'

.loader
    border: 6px solid colors.$blue-dark
    border-top: 6px solid colors.$blue-light
    border-radius: 50%
    width: 48px
    height: 48px
    animation: spinner 2s linear infinite

@keyframes spinner
  0% 
    transform: rotate(0deg)
  100% 
    transform: rotate(360deg)
