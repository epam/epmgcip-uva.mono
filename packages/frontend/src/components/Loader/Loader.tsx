import css from './Loader.module.sass';

interface LoaderProps {
    size?: string;
}

export const Loader = ({ size }: LoaderProps) => {
  return <div className={css.loader} style={{ width: size, height: size }}></div>;
};
