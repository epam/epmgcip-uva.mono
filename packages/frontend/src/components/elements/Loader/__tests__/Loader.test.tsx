import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { Loader } from '../Loader';

describe('Testing: Loader', () => {
  it('should render', () => {
    const { container } = render(<Loader />);

    expect(container).toMatchSnapshot();
  });
});
