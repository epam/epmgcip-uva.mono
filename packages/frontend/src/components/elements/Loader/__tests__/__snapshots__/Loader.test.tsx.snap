// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing: Loader > should render 1`] = `
<div>
  <div
    class="_loader_b24894"
  />
</div>
`;
